manifests/unit.pp

```diff
@@ -22,54 +22,86 @@
   $install_wantedby        = undef,
 ) {
 
-  validate_re($ensure, [ '^present$', '^absent$' ],
-    "systemd::unit::${name}::ensure is invalid and does not match the regex.")
+  include ::systemd
+
+  if is_string($ensure) {
+    validate_re($ensure, [ '^present$', '^absent$' ],
+      "systemd::unit::${name}::ensure does not match the regex.")
+  } else {
+    fail("systemd::unit::${name}::ensure is not a string.")
+  }
 
   validate_absolute_path($systemd_path)
 
-  if $unit_after != undef {
-    validate_string($unit_after)
+  if $unit_after != undef and is_string($unit_after) == false {
+    fail("systemd::${name}::unit::unit_after is not a string")
   }
-  if $unit_before != undef {
-    validate_string($unit_before)
+  if $unit_before != undef and is_string($unit_before) == false {
```

Comment:
what's the difference with the original beside that's a one line condition and you replace the function `validate_string` with `is_string`? Which error message does `validate_string` throw when it's not a string?

Reply:
The same reason as above. It's to ensure the module doesn't fail to compile if invalid data is sent to it.